spyderlib/plugins/configdialog.py

```diff
@@ -903,77 +916,432 @@ def _save_lang(self):
-                      }
-        from spyderlib.utils import syntaxhighlighters
-        assert all([key in fieldnames
-                    for key in syntaxhighlighters.COLOR_SCHEME_KEYS])
-        for tabname in names:
-            cs_group = QGroupBox(_("Color scheme"))
-            cs_layout = QGridLayout()
-            for row, key in enumerate(syntaxhighlighters.COLOR_SCHEME_KEYS):
-                option = "%s/%s" % (tabname, key)
-                value = self.get_option(option)
-                name = fieldnames[key]
-                if is_text_string(value):
-                    label, clayout = self.create_coloredit(name, option,
-                                                           without_layout=True)
-                    label.setAlignment(Qt.AlignRight|Qt.AlignVCenter)
-                    cs_layout.addWidget(label, row+1, 0)
-                    cs_layout.addLayout(clayout, row+1, 1)
-                else:
-                    label, clayout, cb_bold, cb_italic = self.create_scedit(
-                                            name, option, without_layout=True)
-                    label.setAlignment(Qt.AlignRight|Qt.AlignVCenter)
-                    cs_layout.addWidget(label, row+1, 0)
-                    cs_layout.addLayout(clayout, row+1, 1)
-                    cs_layout.addWidget(cb_bold, row+1, 2)
-                    cs_layout.addWidget(cb_italic, row+1, 3)
-            cs_group.setLayout(cs_layout)
-            if tabname in sh.COLOR_SCHEME_NAMES:
-                def_btn = self.create_button(_("Reset to default values"),
-                                         lambda: self.reset_to_default(tabname))
-                tabs.addTab(self.create_tab(cs_group, def_btn), tabname)
-            else:
-                tabs.addTab(self.create_tab(cs_group), tabname)
-        
+        names = [n for n in self.get_option("names")]
+        custom_names = self.get_option("custom_names", [])
+
+        # Widgets
+        about_label = QLabel(_("Here you can edit the color schemes provided "
+                               "by Spyder or create your own ones by using "
+                               "the options provided below.<br>"))
+        edit_button = QPushButton(_("Edit selected"))
+        create_button = QPushButton(_("Create new scheme"))
+        self.delete_button = QPushButton(_("Delete"))
+        self.preview_editor = CodeEditor(self)
+        self.stacked_widget = QStackedWidget(self)
+        self.reset_button = QPushButton(_("Reset"))
+        self.scheme_editor_dialog = SchemeEditor(parent=self,
+                                                 stack=self.stacked_widget)
+
+        # Widget setup
+        self.scheme_choices_dic = {}
+        about_label.setWordWrap(True)
+        schemes_combo = self.create_combobox(_('Scheme:'),
+                                             [('', '')],
+                                             'selected')
+        self.schemes_combo = schemes_combo.combobox

```

Comment:
This is confusing: you have here the `schemes_combo` and `self.schemes_combo` variables, which stand for different things.

Please select a different or more representative for one or the other.